src/rpc/node.cpp

```diff
@@ -42,7 +42,7 @@ static RPCHelpMan setmocktime()
         RPCExamples{""},
         [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
 {
-    if (!Params().IsMockableChain()) {
+    if (Params().GetChainType() != ChainType::REGTEST) {
```

Comment:
Not really convinced this makes sense though; having the `IsMockable` condition in one place (via an easily greppable name), rather than two places seems sensible. OTOH, could easily have that code just be in rpc/node.cpp rather than kernel/chainparams.h, eg

```c++
static bool IsMockable(const CChainParams& params)
{
    return params.GetChainType() == ChainType::REGTEST;
}

if (!IsMockable(Params())) { error...; }
```

May also make sense to add an error in init.cpp if `-mocktime` is specified with a different chain (in which case mocktime activates but you can't ever change it, so things will break pretty quickly), in which case perhaps putting `IsMockable` into src/chainparams.h (rather than kernel/chainparams.h) or similar would make sense.

May want to also replace `Params()` with `EnsureAnyChainman(request.context).GetParams()` to reduce global accesses?